spieleplattform-frontend/src/pages/Game/AllGamesPage/AllGamesPage.tsx

```diff
@@ -1,40 +1,75 @@
 import Game from "../Game";
+import GameConsole from "../GameConsole";
 import {useEffect, useState} from "react";
 import axios from "axios";
 import GameComponent from "../GameComponent/GameComponent";
-import "./AllGamesPage.css"
+import "./AllGamesPage.scss"
 import Error404Page from "../../Error/Error404Page";
 import ErrorPage from "../../Error/ErrorPage";
+import {useSearchParams} from "react-router-dom";
 
 export default function AllGamesPage() {
-    const [games, setGames] = useState<Game[] | null>(null);
-    const [errorStatus, setErrorStatus] = useState(null);
+    const [games, setGames] = useState<Game[] | null>(null)
+    const [gameConsoles, setGameConsoles] = useState<GameConsole[] | null>(null)
+    const [errorStatus, setErrorStatus] = useState(null)
+
+    const [searchParams, setSearchParams] = useSearchParams()
+    const gameConsoleIdValue = Number(searchParams.get('gameConsoleId'))
+    const gameConsoleId = isNaN(gameConsoleIdValue) ? 0 : gameConsoleIdValue
 
     useEffect(() => {
-        axios.get("/game")
+        if (gameConsoleId === 0)
+            setSearchParams({})
+        const gamesApiPath = "/games" + ((gameConsoleId !== 0) ? "?gameConsoleId=" + gameConsoleId : "")
+
+        axios.get("/gameConsoles")
+            .then((response) => {
+                setGameConsoles(response.data)
+            })
+            .catch(error => {
+                setErrorStatus(error.response.status)
+            })
+
+        axios.get(gamesApiPath)
             .then((response) => {
                 setGames(response.data)
             })
             .catch(error => {
-                setErrorStatus(error.response.status);
+                setErrorStatus(error.response.status)
             })
-    }, [])
+    }, [gameConsoleId, setSearchParams])
 
     if (errorStatus)
         return errorStatus === 404 ? <Error404Page/> : <ErrorPage/>
-    return games && (
-        <div className="flex-container">
-            {games.map(game =>
-                <GameComponent
-                    key={game.id}
-                    id={game.id}
-                    name={game.name}
-                    releaseDate={game.releaseDate}
-                    developer={game.developer}
-                    imgUrl={game.imgUrl}
-                    ratings={game.ratings}
-                />
-            )}
-        </div>
+    return games && gameConsoles && (
+        <>
+            <form className="filter-form">
+                <select name="gameConsoles" id="gameConsoles" value={gameConsoleId} onChange={e => {
+                    const value = e.target.value
+                    if (value === "0")
```

Comment:
Grundsätzlich, sind Codestyle-Fragen auch Projektabhängig. Aus meiner Erfahrung, werden aber geschweifte Klammern bein einzeiligen if-Statements nicht weg gelassen (Hatte ich auch weiter oben an einer for-Schleife gesehen, da auch eher nicht). Das Problem ist, dass es sehr schnell passieren kann, dass man einfach eine weitere Zeile einfügt und dann komische Bugs hat, weil man übersieht, dass die Klammern fehlen. In diesem Fall würde ich persönlich wahrscheinlich ein Ternary Statement benutzen, also `setSearchParams(value === 0 ? {} : {gameConsoleId: value})`.

Reply:
Ok dann schreibe ich die klammern lieber hin.